apps/nestjs-api/src/reminder-emails/reminder-emails.service.ts

```diff
@@ -97,6 +103,66 @@ export class ReminderEmailsService {
     return []
   }
 
+  async getUnmatchedMenteesFor45Days() {
```

Comment:
Hey, I've tried for the last hour to wrap my head around this code but must declare defeat and start asking questions. The set unions and differences is very interesting, but tricky to follow (at least for me).

First and foremost I'd like to get this answered:

What precisely does "unmatched mentees for 45 days" mean? I get that the first easy part: we start off with the set of mentees that were approved exactly 45 days ago.

Then, we need to do ... what? Figure out which one of them currently do **not** have an active mentorship?

Reply:
I explained this to Kate above:

"Hey Kate, let me explain how this works:

First we find all approved mentees exactly 45 days ago. But of course some of them have already applied for mentorships during that 45 days period, so the "difference" function takes care of that. It finds the mentees that are approved BUT don't have any mentorship match yet.

The problem is, having a mentorship match doesn't mean they have a running mentorship, because fetching mentorship matches from Salesforce also returns the applications. So the "difference" unfortunately removes the mentees with active applications too. To fix that, I'm using "union" to merge the first set of mentees with the mentees that have "Applied" status mentorship matches.

Hope it's clear."

Let me know if this explains it or if I should go into a bit more details.

Reply:
Judging just by the name of the function, I expect it to...

find all mentees that:
- was approved exactly 45 days ago
- have 0..n (any number) of Matches at stage `APPLIED` or `DECLINED_BY_MENTOR` (implying that they've either tried applying OR that they have applied, but no applications have been successful so far)
- have precisely ZERO Matches at any of the other stages: `ACCEPTED`, `COMPLETED`, `CANCELLED`, `INVALIDATED_AS_OTHER_MENTOR_ACCEPTED`

Is that correct?

Reply:
It's actually hard to understand what the email itself is for. The "spec" in the original ticket isn't really clear:

> "4. Reminder to Unmatched mentees about the platform and new mentors there. (Unmatched mentee is a mentee that was approved, but didn’t apply for a mentor or applied, but was not accepted, 45 days after approval):Priority 4

How do you interpret it?

Reply:
@ericbolikowski This one is answered by Asya and I added the drawing down.